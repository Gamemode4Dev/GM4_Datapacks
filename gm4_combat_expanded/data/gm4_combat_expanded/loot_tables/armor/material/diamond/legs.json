{
  "type": "minecraft:generic",
  "pools": [
    {
      "rolls": 1,
      "entries": [
        {
          "type": "minecraft:item",
          "name": "minecraft:diamond_leggings",
          "functions": [
            {
              "function": "minecraft:set_attributes",
              "modifiers": [
                {
                  "attribute": "generic.armor",
                  "name": "generic.armor",
                  "amount": 6,
                  "operation": "addition",
                  "slot": "legs"
                },
                {
                  "attribute": "generic.armor_toughness",
                  "name": "generic.armor_toughness",
                  "amount": 2,
                  "operation": "addition",
                  "slot": "legs"
                },
                {
                  "attribute": "generic.knockback_resistance",
                  "name": "generic.knockback_resistance",
                  "amount": 0,
                  "operation": "addition",
                  "slot": "legs"
                }
              ]
            },
            {
              "function": "minecraft:set_name",
              "entity": "this",
              "name": [
                {
                  "translate": "%1$s%3427655$s",
                  "with": [
                    [
                      "Unidentified ",
                      {
                        "translate": "item.minecraft.diamond_leggings"
                      }
                    ],
                    {
                      "translate": "item.gm4.combat_expanded.name.unidentified",
                      "with": [
                        {
                          "translate": "item.minecraft.diamond_leggings"
                        }
                      ]
                    }
                  ],
                  "italic": false
                }
              ]
            },
            {
              "function": "minecraft:set_nbt",
              "tag": "{gm4_combat_expanded:{identified:0,armor_clamp:-6,slot:2,name:'{\"italic\":false,\"translate\":\"item.minecraft.diamond_leggings\"}'}}"
            },
            {
              "function": "minecraft:enchant_randomly",
              "enchantments": [
                "minecraft:depth_strider"
              ],
              "conditions": [
                {
                  "condition": "minecraft:reference",
                  "name": "gm4_combat_expanded:mob/modifier/reef"
                },
                {
                  "condition": "minecraft:random_chance",
                  "chance": 0.5
                }
              ]
            },
            {
              "function": "minecraft:enchant_randomly",
              "enchantments": [
                "minecraft:projectile_protection"
              ],
              "conditions": [
                {
                  "condition": "minecraft:reference",
                  "name": "gm4_combat_expanded:mob/modifier/growth"
                },
                {
                  "condition": "minecraft:random_chance",
                  "chance": 0.8
                }
              ]
            },
            {
              "function": "minecraft:enchant_with_levels",
              "levels": {
                "min": {
                  "type": "minecraft:score",
                  "target": {
                    "type": "minecraft:fixed",
                    "name": "$difficulty"
                  },
                  "score": "gm4_ce_data",
                  "scale": 1.5
                },
                "max": {
                  "type": "minecraft:score",
                  "target": {
                    "type": "minecraft:fixed",
                    "name": "$difficulty"
                  },
                  "score": "gm4_ce_data",
                  "scale": 2.5
                }
              },
              "treasure": false,
              "conditions": [
                {
                  "condition": "minecraft:value_check",
                  "value": {
                    "type": "minecraft:score",
                    "target": {
                      "type": "minecraft:fixed",
                      "name": "$difficulty"
                    },
                    "score": "gm4_ce_data"
                  },
                  "range": {
                    "min": 2
                  }
                },
                {
                  "condition": "minecraft:random_chance",
                  "chance": 0.2
                }
              ]
            }
          ]
        }
      ]
    }
  ]
}
