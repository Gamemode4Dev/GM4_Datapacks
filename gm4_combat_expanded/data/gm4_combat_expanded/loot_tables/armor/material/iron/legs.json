{
  "type": "minecraft:generic",
  "pools": [
    {
      "rolls": 1,
      "entries": [
        {
          "type": "minecraft:item",
          "name": "minecraft:iron_leggings",
          "functions": [
            {
              "function": "minecraft:set_attributes",
              "modifiers": [
                {
                  "attribute": "generic.armor",
                  "name": "generic.armor",
                  "id": "D8499B04-0E66-4726-AB29-64469D734E0D",
                  "amount": 5,
                  "operation": "addition",
                  "slot": "legs"
                }
              ]
            },
            {
              "function": "minecraft:set_lore",
              "entity": "this",
              "lore": [
                {
                  "translate": "item.gm4.combat_expanded.lore.unidentified",
                  "fallback": "%s Unidentified Item - Drop to Identify",
                  "with": [
                    {
                      "text": "âš ",
                      "color": "red",
                      "italic": false
                    }
                  ],
                  "color": "dark_red",
                  "italic": false
                }
              ]
            },
            {
              "function": "minecraft:set_nbt",
              "tag": "{gm4_combat_expanded:{durability:225,identified:0,armor_clamp:-5,slot:2,material:3,name:'{\"italic\":false,\"translate\":\"item.minecraft.iron_leggings\"}'}}"
            },
            {
              "function": "minecraft:enchant_randomly",
              "enchantments": [
                "minecraft:projectile_protection"
              ],
              "conditions": [
                {
                  "condition": "minecraft:reference",
                  "name": "gm4_combat_expanded:mob/modifier/growth"
                },
                {
                  "condition": "minecraft:value_check",
                  "value": {
                    "type": "minecraft:binomial",
                    "n": 1,
                    "p": {
                      "type": "minecraft:score",
                      "target": {
                        "type": "minecraft:fixed",
                        "name": "$difficulty"
                      },
                      "score": "gm4_ce_data",
                      "scale": 0.008
                    }
                  },
                  "range": 1
                }
              ]
            },
            {
              "function": "minecraft:enchant_with_levels",
              "levels": {
                "min": {
                  "type": "minecraft:score",
                  "target": {
                    "type": "minecraft:fixed",
                    "name": "$difficulty"
                  },
                  "score": "gm4_ce_data",
                  "scale": 0.15
                },
                "max": {
                  "type": "minecraft:score",
                  "target": {
                    "type": "minecraft:fixed",
                    "name": "$difficulty"
                  },
                  "score": "gm4_ce_data",
                  "scale": 0.25
                }
              },
              "treasure": false,
              "conditions": [
                {
                  "condition": "minecraft:random_chance",
                  "chance": 0.2
                }
              ]
            }
          ]
        }
      ]
    }
  ]
}
