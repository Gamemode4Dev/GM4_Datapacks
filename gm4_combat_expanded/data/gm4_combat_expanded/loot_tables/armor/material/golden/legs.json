{
  "type": "minecraft:generic",
  "pools": [
    {
      "rolls": 1,
      "entries": [
        {
          "type": "minecraft:item",
          "name": "minecraft:golden_leggings",
          "functions": [
            {
              "function": "minecraft:set_attributes",
              "modifiers": [
                {
                  "attribute": "generic.armor",
                  "name": "generic.armor",
                  "id": "D8499B04-0E66-4726-AB29-64469D734E0D",
                  "amount": 3,
                  "operation": "addition",
                  "slot": "legs"
                }
              ]
            },
            {
              "function": "minecraft:set_lore",
              "entity": "this",
              "lore": [
                {
                  "translate": "item.gm4.combat_expanded.lore.unidentified",
                  "fallback": "%s Unidentified Item - Drop to Identify",
                  "with": [
                    {
                      "text": "âš ",
                      "color": "red",
                      "italic": false
                    }
                  ],
                  "color": "dark_red",
                  "italic": false
                }
              ]
            },
            {
              "function": "minecraft:set_nbt",
              "tag": "{gm4_combat_expanded:{durability:105,identified:0,armor_clamp:-3,slot:2,material:4,name:'{\"italic\":false,\"translate\":\"item.minecraft.golden_leggings\"}'}}"
            },
            {
              "function": "minecraft:enchant_randomly",
              "enchantments": [
                "minecraft:projectile_protection"
              ],
              "conditions": [
                {
                  "condition": "minecraft:reference",
                  "name": "gm4_combat_expanded:mob/modifier/growth"
                },
                {
                  "condition": "minecraft:value_check",
                  "value": {
                    "type": "minecraft:binomial",
                    "n": 1,
                    "p": {
                      "type": "minecraft:score",
                      "target": {
                        "type": "minecraft:fixed",
                        "name": "$difficulty"
                      },
                      "score": "gm4_ce_data",
                      "scale": 0.008
                    }
                  },
                  "range": 1
                }
              ]
            },
            {
              "function": "minecraft:enchant_with_levels",
              "levels": {
                "min": {
                  "type": "minecraft:score",
                  "target": {
                    "type": "minecraft:fixed",
                    "name": "$difficulty"
                  },
                  "score": "gm4_ce_data",
                  "scale": 0.2
                },
                "max": {
                  "type": "minecraft:score",
                  "target": {
                    "type": "minecraft:fixed",
                    "name": "$difficulty"
                  },
                  "score": "gm4_ce_data",
                  "scale": 0.3
                }
              },
              "treasure": false,
              "conditions": [
                {
                  "condition": "minecraft:random_chance",
                  "chance": 0.2
                }
              ]
            }
          ]
        }
      ]
    }
  ]
}
